posts/2020-02-14-cats_stm_0.7.0.md

```diff
@@ -0,0 +1,110 @@
+--- title: Cats STM 0.7.0 ---
+
+Today marks the release of version 0.7.0 of [Cats
+STM](https://github.com/TimWSpence/cats-stm) with several significant bug fixes
+and improvements to the fairness of retry scheduling.
+
+If you are not familiar with the concept of STM (Software Transactional Memory)
+then you should definitely read the wonderful original paper [Beautiful
+Concurrency](https://www.microsoft.com/en-us/research/wp-content/uploads/2016/02/beautiful.pdf)
+or at least some of the docs for this library :D However, I will endeavour to
+give a very brief overview here.
+
+The motivating observation is that the traditional tools for doing concurrent
+programming (mutexes, semaphores, etc) do not compose. Given two functions `f: A
+=> B` and `g: B => C` each acquiring mutexes, we cannot reason about the locking
+behaviour of `f andThen g` in the presence of concurrency (see sections 2.1 and 2.2
+of the paper).
+
+The solution to this is the `STM` monad. This is compositional by definition
+(indeed, this practically _is_ the definition of a monad) and exposes combinators
+such as
+
+`def map[A, B](s: STM[A], f: A => B): STM[B]`
+`def orElse[A](attempt: STM[A], fallback: STM[A]): STM[A]`
+`def check(check: => Boolean): STM[Unit]`
+
+How do we obtain a value of type `STM[A]` in the first place? These are returned by
+the operations defined on `TVar`s (transactional vars):
+`get: STM[A]`
```

Comment:
Nitpick but seems like you might want to add `def` here and below with `atomically` to match the signatures given above? Could also use a markdown list to make the formatting nicer.